subsys/net/lib/dns/dns_pack.c

```diff
@@ -404,3 +406,168 @@ int dns_copy_qname(u8_t *buf, u16_t *len, u16_t size,
 
 	return rc;
 }
+
+int dns_unpack_query_header(struct dns_msg_t *msg, u16_t *src_id)
+{
+	u8_t *dns_header;
+	u16_t size;
+	int qdcount;
+	int rc;
+
+	dns_header = msg->msg;
+	size = msg->msg_size;
+
+	if (size < DNS_MSG_HEADER_SIZE) {
+		return -ENOMEM;
+	}
+
+	if (dns_header_qr(dns_header) != DNS_QUERY) {
+		return -EINVAL;
+	}
+
+	if (dns_header_opcode(dns_header) != DNS_QUERY) {
+		return -EINVAL;
+	}
+
+	if (dns_header_z(dns_header) != 0) {
+		return -EINVAL;
+	}
+
+	rc = dns_header_rcode(dns_header);
+	if (rc != 0) {
```

Comment:
you don't seem to use rc anywhere else, so maybe you can get rid of that variable and test dns_header_rcode() directly.